samples/oauth_php/VetsApiClaims/app/Http/Controllers/HomeController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\OauthSession;
+
+class HomeController extends Controller
+{
+    public function getIndex()
+    {
+        $nonce_base = bin2hex(random_bytes(20));
+        session(['nonce_key' => $nonce_base]);
+        session(['login_time' => time()]);
+        $scope = 'openid profile offline_access claim.read claim.write service_history.read disability_rating.read veteran_status.read';
+        $oauth_params = [
+          'client_id' => env('VA_CLIENT_ID'),
+          'nonce' => hash("sha256", $nonce_base . env('VA_CLIENT_SECRET')),
+          'redirect_uri' => "http://localhost:3000/callback",
+          'response_type' => 'code',
+          'scope' => $scope,
+          'state' => session('login_time')
+        ];
+        $oauth_url = env('VETS_API_URL') . "/oauth2/authorization?" . http_build_query($oauth_params);
+        return view('home', ['oauth_url' => $oauth_url]);
+    }
+
+    public function getCallback(Request $request)
+    {
+      if ($request->input('code') == null) {
+        return redirect("/");  
+      };
+      if ((int)$request->input('state') != session('login_time')) {
+        session(['Invalid state' => 'Invalid State']);
+        return redirect("/");
+      }
+      $body = [
+        'grant_type' => 'authorization_code',
+        'code' => $request->input('code'),
+        'state' => $request->input('state'),
+        'redirect_uri' => "http://localhost:3000/callback"
+      ];
+      $url = env('VETS_API_URL') . "/oauth2/token";
+      $curl = curl_init($url);
```

Comment:
same as before, also will allow cleaner testing to mock your Guzzle request object and keep it a bit DRYer as well

Reply:
We decided to use Guzzle on the data endpoints and curl on the oauth flow just to show 2 styles